activesupport/lib/active_support/testing/parallelize_executor.rb

```diff
@@ -45,19 +45,34 @@ def build_parallel_executor
 
         def parallelize
           @parallelized = true
-          Minitest::Test.parallelize_me!
```

Comment:
This line was forcing all test to run serially independently of the `self.test_order` method. 
`parallelize_me!` [now only runs](https://github.com/rails/rails/pull/44938/files#diff-07105cc6673b7fc7f06051a0f50bdfa3c03e3c78ad2c817ddc23be405b3e143aR48) for test suites that have `test_order == :parallel`